### ISSUE TITLE
About the block used in the paper

### ISSUE THREAD

**Issue by 1chijianren** [YonghengSun1997/MSCA-Net#9]
Dear Authors,

I hope this message finds you well. I've been carefully studying your code architecture, and I have a few questions regarding the implementation of specific modules mentioned in your paper.

Module Class Mapping:
I've successfully located the class corresponding to the MSB (Multi-scale Block) module in the codebase. However, I'm having trouble identifying the classes that implement the GL-CSAM (Global-Local Contextual Self-Attention Module) and SADS (Scale-Aware Dynamic Selection) modules. Could you kindly point me to the specific classes or files where these modules are defined?
Unused Classes:
During my review, I noticed that the codebase contains a relatively large number of classes. Are there any classes that are part of the codebase but not actually used in the final architecture or training pipeline? If so, could you provide some guidance on which classes are essential and which ones are for experimental purposes or legacy functionality?

Your insights would be invaluable to my understanding of the implementation details and would greatly assist my research. Thank you for your time and consideration.

Best regards,

**Comment by YonghengSun1997** [comment]
Thank you for your interest in our work and for taking the time to thoroughly examine the codebase. I’m happy to clarify your questions.
1. GL-CSAM Module
The GL-CSAM (Global-Local Contextual Self-Attention Module) is implemented in the lines you referred to:
`attention = self.affinity_attention(c5)
cbam_attn = self.cbam(c5)
c5 = self.gamma1 * attention + self.gamma2 * cbam_attn + self.gamma3 * c5
`
2. SADS Module
The SADS (Scale-Aware Dynamic Selection) part is implemented in:
`dsv43 = self.sw1(dsv4, dsv3)
dsv432 = self.sw2(dsv43, dsv2)
dsv4321 = self.sw3(dsv432, dsv1)
`
3. About Unused Classes
You're right that there are several classes in the codebase. Some of them were designed for ablation studies or earlier experimental versions.

**Comment by 1chijianren** [comment]
Thank you very much!

**Comment by 1chijianren** [comment]
Dear author,
 I would like to ask whether the 400 epochs mentioned in the article refer to the total number across all folds or 80 epochs per fold, which would add up to 500 for 5 folds. Could you please take a moment to clarify this? Thank you.

**Comment by YonghengSun1997** [comment]
400 epochs for each fold.

**Comment by 1chijianren** [comment]
Thank you.And I have another question. If I want to change the dataset, how should I modify this code? Do I just need to change the dataset path?Could you please take a moment to answer this? Thank you.
Wish you a happy life!

**Comment by YonghengSun1997** [comment]
Thank you for your interest!

Yes, if your new dataset has the same format (e.g., image size, label structure, folder organization), you can simply change the dataset path in the configuration or script. However, if the data format differs, you may also need to modify the dataset loading logic in the corresponding Dataset class or data preprocessing scripts.

Feel free to share more details about your dataset if you’re unsure—we’re happy to help.

Wish you a happy life too! 😊

**Comment by 1chijianren** [comment]
Dear Author,

Thank you for your answer. I'm extremely interested in your work and would greatly appreciate it if you could spare a moment to offer some guidance. I'm eager to build upon your research by making improvements, specifically in the areas of handling image boundaries and addressing lesions of varying scales. If you have any insights, recommendations, or directions that you think might be helpful, I would be truly grateful for your input.

Thank you so much for your time and consideration. Even if you're unable to provide suggestions, please know that I deeply appreciate your contributions to the field.

Best regards

**Comment by 1chijianren** [comment]
In addition, I would like to ask whether you used the test sets of the ISIC 2017 and 2016 datasets, or whether you only used the training sets for five-fold cross-validation. Did you use .npy files for all datasets? The ISIC 2016 and 2017 datasets I downloaded from the official website are in .jpg and .png formats, but the ISIC 2018 dataset in the code uses .npy files.

**Comment by 1chijianren** [comment]
Dear Author,

I hope this message finds you well. I've resolved the previous issue, but I encountered a new problem while working with the ISIC2017 dataset. I used the isic-preprocess script from your repository to generate a preprocessed dataset named ISIC2017_npy_all_224_320, which contains .npy files. However, when I tried to run the code, it keeps throwing errors indicating that the images cannot be found.

Here's what I've done:

Uploaded the ISIC2017_npy_all_224_320 dataset to the server.
Modified the code in the main function:
Changed npy_path = os.path.join('./data/ISIC2018_npy_all_224_320/image', name[0])
to the commented-out line:
npy_path = os.path.join('./data/ISIC2017_npy_all_224_320/image', name[0])
Adjusted the argument parser:
Changed parser.add_argument('--root_path', default='./data/ISIC2018_npy_all_224_320', help='root directory of data')
to the commented-out version:
parser.add_argument('--root_path', default='./data/ISIC2017_npy_all_224_320', help='root directory of data')

Despite these changes, the error persists. I'm wondering if there are other places in the code that need to be modified. According to the documentation, the ISIC2016 and ISIC2017 datasets do not use 5-fold cross-validation. Do I need to:

Use a different data loader?
Keep a copy of the ISIC2017_npy_all_224_320 dataset on my local machine?

Could you please provide guidance on how to resolve this issue? Any help would be greatly appreciated.

Best regards，Wish you a happy life!
